### ISSUE TITLE
bounding box resource query

### ISSUE THREAD

**Issue by miguelpeixe** [codigourbano/cidades-comestiveis#14]

**Comment by miguelpeixe** [comment]
returning error:

```
{ [MongoError: n/a]
  name: 'MongoError',
  message: 'n/a',
  '$err': 'Malformed geo query: { $geoWithin: { $geometry: "{"type":"Polygon","coordinates":[[[-46.850166320800774,-23.605520232835943],[-46.850166320800774,-23.479621953329033],[-46.40625,-23.479621953329033],..." } }',
  code: 16677 }
```

**Comment by miguelpeixe** [comment]
@vgeorge any update on this?

**Comment by miguelpeixe** [comment]
I just [fixed](https://github.com/codigourbano/cidades-comestiveis/commit/18a0872b33b013e10882b8065a2f1abd6393683e#diff-587a685e305e547549f045c4e299ed5d) the error, it wasn't parsing the bbox json.

Still, it's returning an empty array (but the count is 1)

**GET api/v1/resources?bbox={"type":"Polygon","coordinates":[[[-46.68790340423584,-23.536527164275974],[-46.68790340423584,-23.524723125388686],[-46.665544509887695,-23.524723125388686],[-46.665544509887695,-23.536527164275974],[-46.68790340423584,-23.536527164275974]]]}**
```json
{"count":1,"perPage":30,"page":1,"resources":[]}
```

**Comment by miguelpeixe** [comment]
still not returning results of the bbox

**Comment by miguelpeixe** [comment]
the code is not parsing the bbox string from the query to JSON, here's a log of the bbox var:

```
[ NaN,
  NaN,
  NaN,
  NaN,
  NaN,
  NaN,
  NaN,
  NaN,
  NaN,
  NaN,
  NaN,
  NaN,
  NaN,
  NaN,
  NaN,
  NaN,
  NaN,
  NaN,
  NaN,
  NaN,
  NaN,
  NaN,
  NaN,
  NaN,
  NaN,
  NaN,
  NaN,
  NaN,
  NaN,
  NaN,
  NaN,
  NaN,
  NaN,
  NaN,
  NaN,
  NaN,
  4,
  6,
  NaN,
  8,
  0,
  6,
  9,
  0,
  7,
  6,
  5,
  3,
  8,
  0,
  8,
  5,
  9,
  NaN,
  NaN,
  2,
  3,
  NaN,
  6,
  4,
  3,
  8,
  9,
  5,
  1,
  9,
  4,
  5,
  2,
  1,
  6,
  9,
  4,
  NaN,
  NaN,
  NaN,
  NaN,
  4,
  6,
  NaN,
  8,
  0,
  6,
  9,
  0,
  7,
  6,
  5,
  3,
  8,
  0,
  8,
  5,
  9,
  NaN,
  NaN,
  2,
  3,
  NaN,
  4,
  5,
  5,
  0,
  5,
  7,
  7,
  5,
  8,
  6,
  9,
  3,
  6,
  NaN,
  NaN,
  NaN,
  NaN,
  4,
  6,
  NaN,
  4,
  6,
  9,
  7,
  6,
  4,
  7,
  0,
  9,
  4,
  7,
  2,
  6,
  5,
  6,
  NaN,
  NaN,
  2,
  3,
  NaN,
  4,
  5,
  5,
  0,
  5,
  7,
  7,
  5,
  8,
  6,
  9,
  3,
  6,
  NaN,
  NaN,
  NaN,
  NaN,
  4,
  6,
  NaN,
  4,
  6,
  9,
  7,
  6,
  4,
  7,
  0,
  9,
  4,
  7,
  2,
  6,
  5,
  6,
  NaN,
  NaN,
  2,
  3,
  NaN,
  6,
  4,
  3,
  8,
  9,
  5,
  1,
  9,
  4,
  5,
  2,
  1,
  6,
  9,
  4,
  NaN,
  NaN,
  NaN,
  NaN,
  4,
  6,
  NaN,
  8,
  0,
  6,
  9,
  0,
  7,
  6,
  5,
  3,
  8,
  0,
  8,
  5,
  9,
  NaN,
  NaN,
  2,
  3,
  NaN,
  6,
  4,
  3,
  8,
  9,
  5,
  1,
  9,
  4,
  5,
  2,
  1,
  6,
  9,
  4,
  NaN,
  NaN,
  NaN,
  NaN ]
```

**Comment by miguelpeixe** [comment]
Here's a proper geojson polygon for reference (what I'm sending as the query param):

```javascript
var bbox = {
  "type": "Polygon",
  "coordinates": [
    [
      [southWest.lng, southWest.lat],
      [southWest.lng, northEast.lat],
      [northEast.lng, northEast.lat],
      [northEast.lng, southWest.lat],
      [southWest.lng, southWest.lat],
    ]
  ]
};
```

**Comment by miguelpeixe** [comment]
I fixed it on 974ec6cbdeb12540287587c58f77ef06394eb57e

You might want to look if I messed up your code a little bit :smile:

**Comment by vgeorge** [comment]
I wasn't using a polygon geojson object but  a bbox parameter like the one described at geojson spec:

http://geojson.org/geojson-spec.html#bounding-boxes

If it is working you I'll keep this way. Later I'll change the docs and tests.
